pyuvdata/tests/test_uvdata.py

```diff
@@ -3909,3 +3914,1014 @@ def test_lsts_from_time_with_only_unique():
+
+    uv_object.downsample_in_time(np.amin(uv_object2.integration_time), blt_order="baseline")
+
+    # increase tolerance on LST if iers.conf.auto_max_age is set to None, as we
+    # do in testing if the iers url is down. See conftest.py for more info.
+    if iers.conf.auto_max_age is None:
+        uv_object._lst_array.tols = (0, 1e-4)
+
+    # make sure that history is correct
+    assert "Upsampled data to 0.939524 second integration time using pyuvdata." in uv_object.history
+    assert "Downsampled data to 1.879048 second integration time using pyuvdata." in uv_object.history
+
+    # overwrite history and check for equality
+    uv_object.history = uv_object2.history
+    assert uv_object == uv_object2
+
+
+@uvtest.skipIf_no_h5py
+@pytest.mark.filterwarnings("ignore:Telescope mock-HERA is not in known_telescopes")
+@pytest.mark.filterwarnings("ignore:There is a gap in the times of baseline")
+def test_resample_in_time():
+    uv_object = UVData()
+    # Note this file has slight variations in the delta t between integrations
+    # that causes our gap test to issue a warning, but the variations are small
+    # We aren't worried about them, so we filter those warnings
+    testfile = os.path.join(DATA_PATH, 'simulated_bda_file.uvh5')
+    uv_object.read(testfile)
+
+    # save some copies for later
+    uv_object2 = uv_object.copy()
+    uv_object3 = uv_object.copy()
+
+    ant_pairs = uv_object.get_antpairs()
+
+    # save some initial info
+    # 2s integration time
+    init_data_1_136 = uv_object.get_data((1, 136))
+    # 4s integration time
+    init_data_1_137 = uv_object.get_data((1, 137))
+    # 8s integration time
+    init_data_1_138 = uv_object.get_data((1, 138))
+    # 16s integration time
+    init_data_136_137 = uv_object.get_data((136, 137))
+
+    uv_object.resample_in_time(8)
+    # Should have all the target integration time
+    assert np.all(np.isclose(uv_object.integration_time, 8))
+
+    # 2s integration time
+    out_data_1_136 = uv_object.get_data((1, 136))
+    # 4s integration time
+    out_data_1_137 = uv_object.get_data((1, 137))
+    # 8s integration time
+    out_data_1_138 = uv_object.get_data((1, 138))
+    # 16s integration time
+    out_data_136_137 = uv_object.get_data((136, 137))
+
+    # check array sizes make sense
+    assert out_data_1_136.size * 4 == init_data_1_136.size
+    assert out_data_1_137.size * 2 == init_data_1_137.size
+    assert out_data_1_138.size == init_data_1_138.size
+    assert out_data_136_137.size / 2 == init_data_136_137.size
+
+    # check some values
+    assert np.isclose(np.mean(init_data_1_136[0:4, 0, 0]), out_data_1_136[0, 0, 0])
+    assert np.isclose(np.mean(init_data_1_137[0:2, 0, 0]), out_data_1_137[0, 0, 0])
+    assert np.isclose(init_data_1_138[0, 0, 0], out_data_1_138[0, 0, 0])
+    assert np.isclose(init_data_136_137[0, 0, 0], out_data_136_137[0, 0, 0])
+
+    # again, with only_downsample set
+    uv_object2.resample_in_time(8, only_downsample=True)
```

Comment:
The only this branch or only that branch of the `resample_in_time` tests could probably be their own tests too.

Reply:
Done.